In [143]:
import pandas as pd
import plotly.express as px

In [144]:
caminho = '../data/processed/df_final.csv'

In [145]:
base = pd.read_csv(caminho, encoding='utf-8', sep='|')

/tmp/ipykernel_5566/2494532192.py:1: DtypeWarning: Columns (6) have mixed types. Specify dtype option on import or set low_memory=False.
  base = pd.read_csv(caminho, encoding='utf-8', sep='|')


In [146]:
round(base.memory_usage(deep=True).sum() / (1024 ** 3), 2)

1.38

In [147]:
base.columns

Index(['Unnamed: 0.1', 'Unnamed: 0', 'NOME_AEROPORTO_ORIGEM',
       'CIDADE_AEROPORTO_ORIGEM', 'UF_AEROPORTO_ORIGEM',
       'SIGLA_ICAO_EMPRESA_AEREA', 'NUMERO_VOO', 'CODIGO_TIPO_LINHA',
       'SIGLA_ICAO_AEROPORTO_ORIGEM', 'SIGLA_ICAO_AEROPORTO_DESTINO',
       'PARTIDA_PREVISTA', 'PARTIDA_REAL', 'CHEGADA_PREVISTA', 'CHEGADA_REAL',
       'SITUACAO_VOO', 'SIGLA_OACI', 'NOME_AEROPORTO_DESTINO',
       'CIDADE_AEROPORTO_DESTINO', 'UF_AEROPORTO_DESTINO', 'NOME_EMPRESA',
       'MES_PARTIDA_PREVISTA', 'NOME_MES_PARTIDA_PREVISTA', 'MES_PARTIDA_REAL',
       'NOME_MES_PARTIDA_REAL', 'MES_CHEGADA_PREVISTA',
       'NOME_MES_CHEGADA_PREVISTA', 'MES_CHEGADA_REAL',
       'NOME_MES_CHEGADA_REAL', 'MINUTOS_ATRASADO_PARTIDA',
       'MINUTOS_ATRASADO_CHEGADA', 'DIA_DA_SEMANA_PARTIDA_PREVISTA',
       'INDICE_SEMANA_PARTIDA_PREVISTA', 'DIA_DA_SEMANA_PARTIDA_REAL',
       'INDICE_SEMANA_PARTIDA_REAL', 'DIA_DA_SEMANA_CHEGADA_REAL',
       'DIA_DA_SEMANA_CHEGADA_PREVISTA', 'INDICE_SEMANA_CHEGADA_R

In [148]:
base.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 717357 entries, 0 to 717356
Data columns (total 40 columns):
 #   Column                          Non-Null Count   Dtype  
---  ------                          --------------   -----  
 0   Unnamed: 0.1                    717357 non-null  int64  
 1   Unnamed: 0                      717357 non-null  int64  
 2   NOME_AEROPORTO_ORIGEM           717357 non-null  object 
 3   CIDADE_AEROPORTO_ORIGEM         717357 non-null  object 
 4   UF_AEROPORTO_ORIGEM             717357 non-null  object 
 5   SIGLA_ICAO_EMPRESA_AEREA        717357 non-null  object 
 6   NUMERO_VOO                      717357 non-null  object 
 7   CODIGO_TIPO_LINHA               717355 non-null  object 
 8   SIGLA_ICAO_AEROPORTO_ORIGEM     717357 non-null  object 
 9   SIGLA_ICAO_AEROPORTO_DESTINO    717357 non-null  object 
 10  PARTIDA_PREVISTA                717357 non-null  object 
 11  PARTIDA_REAL                    717357 non-null  object 
 12  CHEGADA_PREVISTA

In [149]:
base[
    [
        'NOME_AEROPORTO_ORIGEM',
        'CIDADE_AEROPORTO_ORIGEM',
        'UF_AEROPORTO_ORIGEM',
        'SIGLA_ICAO_EMPRESA_AEREA',
        'NUMERO_VOO',
        'CODIGO_TIPO_LINHA',
        'SIGLA_ICAO_AEROPORTO_ORIGEM',
        'SIGLA_ICAO_AEROPORTO_DESTINO',
        'SITUACAO_VOO',
        'SIGLA_OACI',
        'NOME_AEROPORTO_DESTINO',
        'CIDADE_AEROPORTO_DESTINO',
        'UF_AEROPORTO_DESTINO',
        'NOME_EMPRESA',
        'NOME_MES_PARTIDA_PREVISTA',
        'NOME_MES_PARTIDA_REAL',
        'NOME_MES_CHEGADA_PREVISTA',
        'NOME_MES_CHEGADA_REAL',
        'DIA_DA_SEMANA_PARTIDA_PREVISTA',
        'DIA_DA_SEMANA_PARTIDA_REAL',
        'DIA_DA_SEMANA_CHEGADA_REAL',
        'DIA_DA_SEMANA_CHEGADA_PREVISTA',
        'FAIXA_ATRASO_PARTIDA',
        'FAIXA_ATRASO_CHEGADA'
    ]
] = base[
    [
        'NOME_AEROPORTO_ORIGEM',
        'CIDADE_AEROPORTO_ORIGEM',
        'UF_AEROPORTO_ORIGEM',
        'SIGLA_ICAO_EMPRESA_AEREA',
        'NUMERO_VOO',
        'CODIGO_TIPO_LINHA',
        'SIGLA_ICAO_AEROPORTO_ORIGEM',
        'SIGLA_ICAO_AEROPORTO_DESTINO',
        'SITUACAO_VOO',
        'SIGLA_OACI',
        'NOME_AEROPORTO_DESTINO',
        'CIDADE_AEROPORTO_DESTINO',
        'UF_AEROPORTO_DESTINO',
        'NOME_EMPRESA',
        'NOME_MES_PARTIDA_PREVISTA',
        'NOME_MES_PARTIDA_REAL',
        'NOME_MES_CHEGADA_PREVISTA',
        'NOME_MES_CHEGADA_REAL',
        'DIA_DA_SEMANA_PARTIDA_PREVISTA',
        'DIA_DA_SEMANA_PARTIDA_REAL',
        'DIA_DA_SEMANA_CHEGADA_REAL',
        'DIA_DA_SEMANA_CHEGADA_PREVISTA',
        'FAIXA_ATRASO_PARTIDA',
        'FAIXA_ATRASO_CHEGADA'
    ]
].astype('string')

In [150]:
base.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 717357 entries, 0 to 717356
Data columns (total 40 columns):
 #   Column                          Non-Null Count   Dtype  
---  ------                          --------------   -----  
 0   Unnamed: 0.1                    717357 non-null  int64  
 1   Unnamed: 0                      717357 non-null  int64  
 2   NOME_AEROPORTO_ORIGEM           717357 non-null  string 
 3   CIDADE_AEROPORTO_ORIGEM         717357 non-null  string 
 4   UF_AEROPORTO_ORIGEM             717357 non-null  string 
 5   SIGLA_ICAO_EMPRESA_AEREA        717357 non-null  string 
 6   NUMERO_VOO                      717357 non-null  string 
 7   CODIGO_TIPO_LINHA               717355 non-null  string 
 8   SIGLA_ICAO_AEROPORTO_ORIGEM     717357 non-null  string 
 9   SIGLA_ICAO_AEROPORTO_DESTINO    717357 non-null  string 
 10  PARTIDA_PREVISTA                717357 non-null  object 
 11  PARTIDA_REAL                    717357 non-null  object 
 12  CHEGADA_PREVISTA

In [151]:
round(base.memory_usage(deep=True).sum() / (1024 ** 3), 2)

1.38

<p>Converter para float</p>
<li> MINUTOS_ATRASADO_PARTIDA        717357 non-null  float64</li>
 <li>  MINUTOS_ATRASADO_CHEGADA        717357 non-null  float64</li>

In [152]:
base[
    [
        'MINUTOS_ATRASADO_PARTIDA',
        'MINUTOS_ATRASADO_CHEGADA'
    ]
] = base[
    [
        'MINUTOS_ATRASADO_PARTIDA',
        'MINUTOS_ATRASADO_CHEGADA'
    ]
].astype('float32')

In [153]:
round(base.memory_usage(deep=True).sum() / (1024 ** 3), 2)

1.38

In [154]:
base.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 717357 entries, 0 to 717356
Data columns (total 40 columns):
 #   Column                          Non-Null Count   Dtype  
---  ------                          --------------   -----  
 0   Unnamed: 0.1                    717357 non-null  int64  
 1   Unnamed: 0                      717357 non-null  int64  
 2   NOME_AEROPORTO_ORIGEM           717357 non-null  string 
 3   CIDADE_AEROPORTO_ORIGEM         717357 non-null  string 
 4   UF_AEROPORTO_ORIGEM             717357 non-null  string 
 5   SIGLA_ICAO_EMPRESA_AEREA        717357 non-null  string 
 6   NUMERO_VOO                      717357 non-null  string 
 7   CODIGO_TIPO_LINHA               717355 non-null  string 
 8   SIGLA_ICAO_AEROPORTO_ORIGEM     717357 non-null  string 
 9   SIGLA_ICAO_AEROPORTO_DESTINO    717357 non-null  string 
 10  PARTIDA_PREVISTA                717357 non-null  object 
 11  PARTIDA_REAL                    717357 non-null  object 
 12  CHEGADA_PREVISTA

<p>Converter para Tempo</p>
<ul>
    <li> 10  PARTIDA_PREVISTA                717357 non-null  object </li>
    <li> 11  PARTIDA_REAL                    717357 non-null  object </li>
    <li>12  CHEGADA_PREVISTA                717357 non-null  object </li>
    <li>13  CHEGADA_REAL                    717357 non-null  object </li>
</ul>

In [155]:
colunas = [
        'PARTIDA_PREVISTA',
        'PARTIDA_REAL',
        'CHEGADA_PREVISTA',
        'CHEGADA_REAL'
    ]

In [156]:
for coluna in colunas:
    base[coluna] = pd.to_datetime(base[coluna], format='%Y-%m-%d %H:%M:%S')

In [157]:
base.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 717357 entries, 0 to 717356
Data columns (total 40 columns):
 #   Column                          Non-Null Count   Dtype         
---  ------                          --------------   -----         
 0   Unnamed: 0.1                    717357 non-null  int64         
 1   Unnamed: 0                      717357 non-null  int64         
 2   NOME_AEROPORTO_ORIGEM           717357 non-null  string        
 3   CIDADE_AEROPORTO_ORIGEM         717357 non-null  string        
 4   UF_AEROPORTO_ORIGEM             717357 non-null  string        
 5   SIGLA_ICAO_EMPRESA_AEREA        717357 non-null  string        
 6   NUMERO_VOO                      717357 non-null  string        
 7   CODIGO_TIPO_LINHA               717355 non-null  string        
 8   SIGLA_ICAO_AEROPORTO_ORIGEM     717357 non-null  string        
 9   SIGLA_ICAO_AEROPORTO_DESTINO    717357 non-null  string        
 10  PARTIDA_PREVISTA                717357 non-null  datetim

In [158]:
round(base.memory_usage(deep=True).sum() / (1024 ** 3), 2)

1.2

<p>Converter para Int</p>
<ul>
    <li> 10  MES_PARTIDA_PREVISTA                717357 non-null  object </li>
    <li> 11  MES_PARTIDA_REAL                    717357 non-null  object </li>
    <li>12  MES_CHEGADA_PREVISTA                717357 non-null  object </li>
    <li>13  CHEGADA_REAL                    717357 non-null  object </li>
    <li> MES_CHEGADA_REAL </li>
    <li>INDICE_SEMANA_PARTIDA_PREVISTA</li>
    <li>INDICE_SEMANA_PARTIDA_REAL</li>
    <li>INDICE_SEMANA_CHEGADA_REAL</li>
    <li>INDICE_SEMANA_CHEGADA_PREVISTA</li>
</ul>

In [159]:
colunas = [
    'MES_PARTIDA_PREVISTA',
    'MES_PARTIDA_REAL',
    'MES_CHEGADA_PREVISTA',
    'MES_CHEGADA_REAL',
    'INDICE_SEMANA_PARTIDA_PREVISTA',
    'INDICE_SEMANA_PARTIDA_REAL',
    'INDICE_SEMANA_CHEGADA_REAL',
    'INDICE_SEMANA_CHEGADA_PREVISTA'
]

In [160]:
for coluna in colunas:
    print(coluna)
    base[coluna] = base[coluna].astype('int32')

MES_PARTIDA_PREVISTA
MES_PARTIDA_REAL
MES_CHEGADA_PREVISTA
MES_CHEGADA_REAL
INDICE_SEMANA_PARTIDA_PREVISTA
INDICE_SEMANA_PARTIDA_REAL
INDICE_SEMANA_CHEGADA_REAL
INDICE_SEMANA_CHEGADA_PREVISTA


In [161]:
round(base.memory_usage(deep=True).sum() / (1024 ** 3), 2)

1.18

In [162]:
base.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 717357 entries, 0 to 717356
Data columns (total 40 columns):
 #   Column                          Non-Null Count   Dtype         
---  ------                          --------------   -----         
 0   Unnamed: 0.1                    717357 non-null  int64         
 1   Unnamed: 0                      717357 non-null  int64         
 2   NOME_AEROPORTO_ORIGEM           717357 non-null  string        
 3   CIDADE_AEROPORTO_ORIGEM         717357 non-null  string        
 4   UF_AEROPORTO_ORIGEM             717357 non-null  string        
 5   SIGLA_ICAO_EMPRESA_AEREA        717357 non-null  string        
 6   NUMERO_VOO                      717357 non-null  string        
 7   CODIGO_TIPO_LINHA               717355 non-null  string        
 8   SIGLA_ICAO_AEROPORTO_ORIGEM     717357 non-null  string        
 9   SIGLA_ICAO_AEROPORTO_DESTINO    717357 non-null  string        
 10  PARTIDA_PREVISTA                717357 non-null  datetim

<p>Converter para Categórica</p>
<ul>
    <li>MES_PARTIDA_PREVISTA</li>
    <li>MES_PARTIDA_REAL</li>
    <li>MES_CHEGADA_PREVISTA</li>
    <li>MES_CHEGADA_REAL</li>
    <li>SIGLA_OACI</li>
</ul>

In [163]:
colunas = [
    'NOME_MES_PARTIDA_PREVISTA', 
    'NOME_MES_PARTIDA_REAL', 
    'NOME_MES_CHEGADA_PREVISTA', 
    'NOME_MES_CHEGADA_REAL', 
    'DIA_DA_SEMANA_PARTIDA_PREVISTA',
    'DIA_DA_SEMANA_PARTIDA_REAL',
    'DIA_DA_SEMANA_CHEGADA_PREVISTA',
    'DIA_DA_SEMANA_CHEGADA_REAL',
    'SITUACAO_VOO',
    'CODIGO_TIPO_LINHA',
    'FAIXA_ATRASO_PARTIDA',
    'FAIXA_ATRASO_CHEGADA'
    ]

In [164]:
for coluna in colunas:
    print(coluna)
    base[coluna] = pd.Categorical(base[coluna])

NOME_MES_PARTIDA_PREVISTA
NOME_MES_PARTIDA_REAL
NOME_MES_CHEGADA_PREVISTA
NOME_MES_CHEGADA_REAL
DIA_DA_SEMANA_PARTIDA_PREVISTA
DIA_DA_SEMANA_PARTIDA_REAL
DIA_DA_SEMANA_CHEGADA_PREVISTA
DIA_DA_SEMANA_CHEGADA_REAL
SITUACAO_VOO
CODIGO_TIPO_LINHA
FAIXA_ATRASO_PARTIDA
FAIXA_ATRASO_CHEGADA


In [165]:
base.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 717357 entries, 0 to 717356
Data columns (total 40 columns):
 #   Column                          Non-Null Count   Dtype         
---  ------                          --------------   -----         
 0   Unnamed: 0.1                    717357 non-null  int64         
 1   Unnamed: 0                      717357 non-null  int64         
 2   NOME_AEROPORTO_ORIGEM           717357 non-null  string        
 3   CIDADE_AEROPORTO_ORIGEM         717357 non-null  string        
 4   UF_AEROPORTO_ORIGEM             717357 non-null  string        
 5   SIGLA_ICAO_EMPRESA_AEREA        717357 non-null  string        
 6   NUMERO_VOO                      717357 non-null  string        
 7   CODIGO_TIPO_LINHA               717355 non-null  category      
 8   SIGLA_ICAO_AEROPORTO_ORIGEM     717357 non-null  string        
 9   SIGLA_ICAO_AEROPORTO_DESTINO    717357 non-null  string        
 10  PARTIDA_PREVISTA                717357 non-null  datetim

In [166]:
round(base.memory_usage(deep=True).sum() / (1024 ** 2), 2)

618.61

In [167]:
base['NOME_MES_CHEGADA_PREVISTA'].unique()

['janeiro', 'fevereiro', 'março', 'abril', 'maio', ..., 'agosto', 'setembro', 'outubro', 'novembro', 'dezembro']
Length: 12
Categories (12, string): [abril, agosto, dezembro, fevereiro, ..., março, novembro, outubro, setembro]

In [175]:
colunas = ['NOME_AEROPORTO_ORIGEM',
       'CIDADE_AEROPORTO_ORIGEM', 'UF_AEROPORTO_ORIGEM',
       'SIGLA_ICAO_EMPRESA_AEREA', 'NUMERO_VOO', 'CODIGO_TIPO_LINHA',
       'SIGLA_ICAO_AEROPORTO_ORIGEM', 'SIGLA_ICAO_AEROPORTO_DESTINO',
       'PARTIDA_PREVISTA', 'PARTIDA_REAL', 'CHEGADA_PREVISTA', 'CHEGADA_REAL',
       'SITUACAO_VOO', 'SIGLA_OACI', 'NOME_AEROPORTO_DESTINO',
       'CIDADE_AEROPORTO_DESTINO', 'UF_AEROPORTO_DESTINO', 'NOME_EMPRESA',
       'MES_PARTIDA_PREVISTA', 'NOME_MES_PARTIDA_PREVISTA', 'MES_PARTIDA_REAL',
       'NOME_MES_PARTIDA_REAL', 'MES_CHEGADA_PREVISTA',
       'NOME_MES_CHEGADA_PREVISTA', 'MES_CHEGADA_REAL',
       'NOME_MES_CHEGADA_REAL', 'MINUTOS_ATRASADO_PARTIDA',
       'MINUTOS_ATRASADO_CHEGADA', 'DIA_DA_SEMANA_PARTIDA_PREVISTA',
       'INDICE_SEMANA_PARTIDA_PREVISTA', 'DIA_DA_SEMANA_PARTIDA_REAL',
       'INDICE_SEMANA_PARTIDA_REAL', 'DIA_DA_SEMANA_CHEGADA_REAL',
       'DIA_DA_SEMANA_CHEGADA_PREVISTA', 'INDICE_SEMANA_CHEGADA_REAL',
       'INDICE_SEMANA_CHEGADA_PREVISTA', 'FAIXA_ATRASO_PARTIDA',
       'FAIXA_ATRASO_CHEGADA']


In [169]:
base = base[colunas]

In [170]:
base.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 717357 entries, 0 to 717356
Data columns (total 38 columns):
 #   Column                          Non-Null Count   Dtype         
---  ------                          --------------   -----         
 0   NOME_AEROPORTO_ORIGEM           717357 non-null  string        
 1   CIDADE_AEROPORTO_ORIGEM         717357 non-null  string        
 2   UF_AEROPORTO_ORIGEM             717357 non-null  string        
 3   SIGLA_ICAO_EMPRESA_AEREA        717357 non-null  string        
 4   NUMERO_VOO                      717357 non-null  string        
 5   CODIGO_TIPO_LINHA               717355 non-null  category      
 6   SIGLA_ICAO_AEROPORTO_ORIGEM     717357 non-null  string        
 7   SIGLA_ICAO_AEROPORTO_DESTINO    717357 non-null  string        
 8   PARTIDA_PREVISTA                717357 non-null  datetime64[ns]
 9   PARTIDA_REAL                    717357 non-null  datetime64[ns]
 10  CHEGADA_PREVISTA                717357 non-null  datetim

In [171]:
round(base.memory_usage(deep=True).sum() / (1024 ** 2), 2)

607.66

In [174]:
for coluna in base.columns:
    print(coluna)
    print(base[coluna].unique())
    print()
    print()
    print()

NOME_AEROPORTO_ORIGEM
<StringArray>
[                     'BAURU/AREALVA',                        'SANTA MARIA',
             'ESTADUAL DARIO GUARITA',    'PRESIDENTE JUSCELINO KUBITSCHEK',
          'ADALBERTO MENDES DA SILVA',                     'TANCREDO NEVES',
                       'CAMPO GRANDE',              'SERAFIN ENOSS BERTASO',
                        'AFONSO PENA',                    'MARECHAL RONDON',
 ...
               'VITÓRIA DA CONQUISTA',                'MÁRIO PEREIRA LOPES',
 'PRESIDENTE JOÃO BATISTA FIGUEIREDO',                         'TERRAVISTA',
           'JUVENAL LOUREIRO CARDOSO',                          'ARAGUAÍNA',
                              'URUCU',                           'CARAUARI',
                              'COARI',                       'BARRA GRANDE']
Length: 109, dtype: string



CIDADE_AEROPORTO_ORIGEM
<StringArray>
[             'AREALVA',              'ARACAJU',            'ARAÇATUBA',
             'BRASÍLIA',             'CASCAVEL',

In [176]:
particao = [
    'MES_PARTIDA_PREVISTA',
    'MES_PARTIDA_REAL',
    'MES_CHEGADA_PREVISTA',
    'MES_CHEGADA_REAL',
    'INDICE_SEMANA_PARTIDA_PREVISTA',
    'INDICE_SEMANA_PARTIDA_REAL',
    'INDICE_SEMANA_CHEGADA_PREVISTA',
    'UF_AEROPORTO_ORIGEM',
    'UF_AEROPORTO_DESTINO',
    'CODIGO_TIPO_LINHA',
    'SIGLA_ICAO_AEROPORTO_ORIGEM',
    'SIGLA_ICAO_AEROPORTO_DESTINO',
    'SIGLA_ICAO_EMPRESA_AEREA',
    'SITUACAO_VOO',
    'SIGLA_OACI',
    'FAIXA_ATRASO_PARTIDA',
    'FAIXA_ATRASO_CHEGADA'
]

In [178]:
base[colunas].to_parquet('../data/processed/df_final.parquet')

- abir parquet

In [188]:
base = pd.read_parquet('../data/processed/df_final.parquet')

In [189]:
base.head()

NOME_AEROPORTO_ORIGEM CIDADE_AEROPORTO_ORIGEM UF_AEROPORTO_ORIGEM  \
0         BAURU/AREALVA                 AREALVA                  SP   
1         BAURU/AREALVA                 AREALVA                  SP   
2         BAURU/AREALVA                 AREALVA                  SP   
3         BAURU/AREALVA                 AREALVA                  SP   
4         BAURU/AREALVA                 AREALVA                  SP   

  SIGLA_ICAO_EMPRESA_AEREA NUMERO_VOO CODIGO_TIPO_LINHA  \
0                      AZU       2455                 N   
1                      AZU       2455                 N   
2                      AZU       2455                 N   
3                      AZU       2455                 N   
4                      AZU       2455                 N   

  SIGLA_ICAO_AEROPORTO_ORIGEM SIGLA_ICAO_AEROPORTO_DESTINO  \
0                        SBAE                         SBPS   
1                        SBAE                         SBPS   
2                        SBAE                         SBPS   
3                        SBAE                         SBPS   
4                        SBAE                         SBPS   

     PARTIDA_PREVISTA        PARTIDA_REAL  ... DIA_DA_SEMANA_PARTIDA_PREVISTA  \
0 2022-01-02 00:05:00 2022-01-01 23:35:00  ...                        Domingo   
1 2022-01-09 01:00:00 2022-01-09 00:48:00  ...                        Domingo   
2 2022-01-23 11:05:00 2022-01-23 10:54:00  ...                        Domingo   
3 2022-01-30 11:05:00 2022-01-30 11:01:00  ...                        Domingo   
4 2022-02-26 23:00:00 2022-02-26 22:56:00  ...                         Sábado   

  INDICE_SEMANA_PARTIDA_PREVISTA DIA_DA_SEMANA_PARTIDA_REAL  \
0                              6                     Sábado   
1                              6                    Domingo   
2                              6                    Domingo   
3                              6                    Domingo   
4                              5                     Sábado   

  INDICE_SEMANA_PARTIDA_REAL DIA_DA_SEMANA_CHEGADA_REAL  \
0                          5                    Domingo   
1                          6                    Domingo   
2                          6                    Domingo   
3                          6                    Domingo   
4                          5                    Domingo   

  DIA_DA_SEMANA_CHEGADA_PREVISTA INDICE_SEMANA_CHEGADA_REAL  \
0                        Domingo                          6   
1                        Domingo                          6   
2                        Domingo                          6   
3                        Domingo                          6   
4                        Domingo                          6   

  INDICE_SEMANA_CHEGADA_PREVISTA  FAIXA_ATRASO_PARTIDA FAIXA_ATRASO_CHEGADA  
0                              6            No horário           No horário  
1                              6            No horário           No horário  
2                              6            No horário           No horário  
3                              6            No horário           No horário  
4                              6            No horário           No horário  

[5 rows x 38 columns]

In [191]:
base.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 717357 entries, 0 to 717356
Data columns (total 38 columns):
 #   Column                          Non-Null Count   Dtype         
---  ------                          --------------   -----         
 0   NOME_AEROPORTO_ORIGEM           717357 non-null  string        
 1   CIDADE_AEROPORTO_ORIGEM         717357 non-null  string        
 2   UF_AEROPORTO_ORIGEM             717357 non-null  string        
 3   SIGLA_ICAO_EMPRESA_AEREA        717357 non-null  string        
 4   NUMERO_VOO                      717357 non-null  string        
 5   CODIGO_TIPO_LINHA               717355 non-null  category      
 6   SIGLA_ICAO_AEROPORTO_ORIGEM     717357 non-null  string        
 7   SIGLA_ICAO_AEROPORTO_DESTINO    717357 non-null  string        
 8   PARTIDA_PREVISTA                717357 non-null  datetime64[ns]
 9   PARTIDA_REAL                    717357 non-null  datetime64[ns]
 10  CHEGADA_PREVISTA                717357 non-null  datetim

In [182]:
base['SIGLA_ICAO_EMPRESA_AEREA'].value_counts()

SIGLA_ICAO_EMPRESA_AEREA
AZU    282255
TAM    211680
GLO    196739
PTB     14369
PAM      5090
TTL      2914
MWM      1503
LTG      1403
CLX       308
TAP       246
GEC       214
TPA       207
LAE       181
ABJ       100
LCO        69
LAN        38
GTI         9
LPE         8
ARG         6
LAP         3
DLH         2
AAL         2
AFR         2
BAW         2
RZO         1
CKS         1
UPS         1
AVA         1
QTR         1
KLM         1
AEA         1
Name: count, dtype: Int64

- Situação Vôo

In [196]:
colunas = ['SIGLA_ICAO_EMPRESA_AEREA', 'SITUACAO_VOO', 'MES_PARTIDA_PREVISTA', 'NOME_MES_PARTIDA_PREVISTA']
teste = pd.read_parquet('../data/processed/df_final.parquet', columns=colunas)

In [197]:
teste.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 717357 entries, 0 to 717356
Data columns (total 4 columns):
 #   Column                     Non-Null Count   Dtype   
---  ------                     --------------   -----   
 0   SIGLA_ICAO_EMPRESA_AEREA   717357 non-null  string  
 1   SITUACAO_VOO               717357 non-null  category
 2   MES_PARTIDA_PREVISTA       717357 non-null  int32   
 3   NOME_MES_PARTIDA_PREVISTA  717357 non-null  category
dtypes: category(2), int32(1), string(1)
memory usage: 9.6 MB


In [198]:
teste.head()

SIGLA_ICAO_EMPRESA_AEREA SITUACAO_VOO  MES_PARTIDA_PREVISTA  \
0                      AZU    REALIZADO                     1   
1                      AZU    REALIZADO                     1   
2                      AZU    REALIZADO                     1   
3                      AZU    REALIZADO                     1   
4                      AZU    REALIZADO                     2   

  NOME_MES_PARTIDA_PREVISTA  
0                   janeiro  
1                   janeiro  
2                   janeiro  
3                   janeiro  
4                 fevereiro

In [199]:
total = base.query(
    'SIGLA_ICAO_EMPRESA_AEREA == "TAM"  ' \
    ' and SIGLA_ICAO_AEROPORTO_ORIGEM == "SBRP"'\
    ' and MES_PARTIDA_PREVISTA in [11,12] '
)
total

NOME_AEROPORTO_ORIGEM CIDADE_AEROPORTO_ORIGEM UF_AEROPORTO_ORIGEM  \
537999           LEITE LOPES          RIBEIRÃO PRETO                  SP   
538000           LEITE LOPES          RIBEIRÃO PRETO                  SP   
538001           LEITE LOPES          RIBEIRÃO PRETO                  SP   
538002           LEITE LOPES          RIBEIRÃO PRETO                  SP   
538003           LEITE LOPES          RIBEIRÃO PRETO                  SP   
...                      ...                     ...                 ...   
538109           LEITE LOPES          RIBEIRÃO PRETO                  SP   
538111           LEITE LOPES          RIBEIRÃO PRETO                  SP   
538112           LEITE LOPES          RIBEIRÃO PRETO                  SP   
538113           LEITE LOPES          RIBEIRÃO PRETO                  SP   
538114           LEITE LOPES          RIBEIRÃO PRETO                  SP   

       SIGLA_ICAO_EMPRESA_AEREA NUMERO_VOO CODIGO_TIPO_LINHA  \
537999                      TAM       3433                 N   
538000                      TAM       3433                 N   
538001                      TAM       3867                 N   
538002                      TAM       3433                 N   
538003                      TAM       3867                 N   
...                         ...        ...               ...   
538109                      TAM       3433                 N   
538111                      TAM       3867                 N   
538112                      TAM       3433                 N   
538113                      TAM       3867                 N   
538114                      TAM       3867                 N   

       SIGLA_ICAO_AEROPORTO_ORIGEM SIGLA_ICAO_AEROPORTO_DESTINO  \
537999                        SBRP                         SBGR   
538000                        SBRP                         SBGR   
538001                        SBRP                         SBGR   
538002                        SBRP                         SBGR   
538003                        SBRP                         SBGR   
...                            ...                          ...   
538109                        SBRP                         SBGR   
538111                        SBRP                         SBGR   
538112                        SBRP                         SBGR   
538113                        SBRP                         SBGR   
538114                        SBRP                         SBGR   

          PARTIDA_PREVISTA        PARTIDA_REAL  ...  \
537999 2022-11-01 09:30:00 2022-11-01 09:30:00  ...   
538000 2022-11-02 09:30:00 2022-11-02 09:30:00  ...   
538001 2022-11-02 14:20:00 2022-11-02 14:17:00  ...   
538002 2022-11-03 09:30:00 2022-11-03 09:25:00  ...   
538003 2022-11-03 14:20:00 2022-11-03 14:12:00  ...   
...                    ...                 ...  ...   
538109 2022-12-29 09:30:00 2022-12-29 09:30:00  ...   
538111 2022-12-29 14:20:00 2022-12-29 14:11:00  ...   
538112 2022-12-30 09:30:00 2022-12-30 09:29:00  ...   
538113 2022-12-30 14:00:00 2022-12-30 13:50:00  ...   
538114 2022-12-31 14:20:00 2022-12-31 14:11:00  ...   

       DIA_DA_SEMANA_PARTIDA_PREVISTA INDICE_SEMANA_PARTIDA_PREVISTA  \
537999                    Terça-feira                              1   
538000                   Quarta-feira                              2   
538001                   Quarta-feira                              2   
538002                   Quinta-feira                              3   
538003                   Quinta-feira                              3   
...                               ...                            ...   
538109                   Quinta-feira                              3   
538111                   Quinta-feira                              3   
538112                    Sexta-feira                              4   
538113                    Sexta-feira                              4   
538114                         Sábado                              

In [200]:
total = base.query(
    'SIGLA_ICAO_EMPRESA_AEREA == "TAM"  ' \
    ' and SIGLA_ICAO_AEROPORTO_ORIGEM == "SBRP"'\
    ' and MES_PARTIDA_PREVISTA in [11] '
)[
    [
        'NOME_MES_PARTIDA_PREVISTA',
        'SIGLA_ICAO_EMPRESA_AEREA', 
        'MES_PARTIDA_PREVISTA',
        'SITUACAO_VOO'
    ]
].value_counts()
# teste = base.query('SIGLA_ICAO_EMPRESA_AEREA == "TAP"')[['SIGLA_ICAO_EMPRESA_AEREA', 'SITUACAO_VOO']].value_counts( normalize=True) * 100
total


NOME_MES_PARTIDA_PREVISTA  SIGLA_ICAO_EMPRESA_AEREA  MES_PARTIDA_PREVISTA  SITUACAO_VOO
novembro                   TAM                       11                    REALIZADO       53
                                                                           CANCELADO        2
Name: count, dtype: int64

In [201]:
percentual = base.query(
    'SIGLA_ICAO_EMPRESA_AEREA == "TAM"  ' \
    ' and SIGLA_ICAO_AEROPORTO_ORIGEM == "SBRP"' \
    ' and MES_PARTIDA_PREVISTA in [11]  '
    )[
        [
            'NOME_MES_PARTIDA_PREVISTA',
            'SIGLA_ICAO_EMPRESA_AEREA', 
            'MES_PARTIDA_PREVISTA',
            'SITUACAO_VOO'
        ]
    ].value_counts( normalize=True) * 100
percentual

NOME_MES_PARTIDA_PREVISTA  SIGLA_ICAO_EMPRESA_AEREA  MES_PARTIDA_PREVISTA  SITUACAO_VOO
novembro                   TAM                       11                    REALIZADO       96.363636
                                                                           CANCELADO        3.636364
Name: proportion, dtype: float64

In [205]:
df_resultado = pd.concat([total, percentual], axis=1).reset_index()
df_resultado

NOME_MES_PARTIDA_PREVISTA SIGLA_ICAO_EMPRESA_AEREA  MES_PARTIDA_PREVISTA  \
0                  novembro                      TAM                    11   
1                  novembro                      TAM                    11   

  SITUACAO_VOO  count  proportion  
0    REALIZADO     53   96.363636  
1    CANCELADO      2    3.636364

In [206]:
df_resultado.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2 entries, 0 to 1
Data columns (total 6 columns):
 #   Column                     Non-Null Count  Dtype   
---  ------                     --------------  -----   
 0   NOME_MES_PARTIDA_PREVISTA  2 non-null      category
 1   SIGLA_ICAO_EMPRESA_AEREA   2 non-null      string  
 2   MES_PARTIDA_PREVISTA       2 non-null      int32   
 3   SITUACAO_VOO               2 non-null      category
 4   count                      2 non-null      int64   
 5   proportion                 2 non-null      float64 
dtypes: category(2), float64(1), int32(1), int64(1), string(1)
memory usage: 716.0 bytes


In [208]:
df_resultado['count'] = df_resultado['count'].astype('int32')
df_resultado['proportion'] = df_resultado['proportion'].astype('float32')

In [209]:
df_resultado.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2 entries, 0 to 1
Data columns (total 6 columns):
 #   Column                     Non-Null Count  Dtype   
---  ------                     --------------  -----   
 0   NOME_MES_PARTIDA_PREVISTA  2 non-null      category
 1   SIGLA_ICAO_EMPRESA_AEREA   2 non-null      string  
 2   MES_PARTIDA_PREVISTA       2 non-null      int32   
 3   SITUACAO_VOO               2 non-null      category
 4   count                      2 non-null      int32   
 5   proportion                 2 non-null      float32 
dtypes: category(2), float32(1), int32(2), string(1)
memory usage: 700.0 bytes


- Desempenho da Situação do Voo

In [212]:
colunas = ['SIGLA_ICAO_EMPRESA_AEREA', 'SITUACAO_VOO', 'MES_PARTIDA_PREVISTA', 'NOME_MES_PARTIDA_PREVISTA', 'SIGLA_ICAO_AEROPORTO_ORIGEM']
teste = pd.read_parquet('../data/processed/df_final.parquet', columns=colunas)

In [213]:
teste = teste.query('SIGLA_ICAO_EMPRESA_AEREA == "TAM"  ' \
            ' and SIGLA_ICAO_AEROPORTO_ORIGEM == "SBRP"'
            ' and SITUACAO_VOO == "REALIZADO" ' 
            )

teste.head()

SIGLA_ICAO_EMPRESA_AEREA SITUACAO_VOO  MES_PARTIDA_PREVISTA  \
488380                      TAM    REALIZADO                     1   
488381                      TAM    REALIZADO                     1   
537779                      TAM    REALIZADO                     1   
537780                      TAM    REALIZADO                     1   
537781                      TAM    REALIZADO                     1   

       NOME_MES_PARTIDA_PREVISTA SIGLA_ICAO_AEROPORTO_ORIGEM  
488380                   janeiro                        SBRP  
488381                   janeiro                        SBRP  
537779                   janeiro                        SBRP  
537780                   janeiro                        SBRP  
537781                   janeiro                        SBRP

In [218]:
teste.info()

<class 'pandas.core.frame.DataFrame'>
Index: 345 entries, 488380 to 680870
Data columns (total 5 columns):
 #   Column                       Non-Null Count  Dtype   
---  ------                       --------------  -----   
 0   SIGLA_ICAO_EMPRESA_AEREA     345 non-null    string  
 1   SITUACAO_VOO                 345 non-null    category
 2   MES_PARTIDA_PREVISTA         345 non-null    int32   
 3   NOME_MES_PARTIDA_PREVISTA    345 non-null    category
 4   SIGLA_ICAO_AEROPORTO_ORIGEM  345 non-null    string  
dtypes: category(2), int32(1), string(2)
memory usage: 10.6 KB


In [227]:
teste['SITUACAO_VOO'] = teste['SITUACAO_VOO'].astype('string')
teste['NOME_MES_PARTIDA_PREVISTA'] = teste['NOME_MES_PARTIDA_PREVISTA'].astype('string')
teste.info()

<class 'pandas.core.frame.DataFrame'>
Index: 345 entries, 488380 to 680870
Data columns (total 5 columns):
 #   Column                       Non-Null Count  Dtype 
---  ------                       --------------  ----- 
 0   SIGLA_ICAO_EMPRESA_AEREA     345 non-null    string
 1   SITUACAO_VOO                 345 non-null    string
 2   MES_PARTIDA_PREVISTA         345 non-null    int32 
 3   NOME_MES_PARTIDA_PREVISTA    345 non-null    string
 4   SIGLA_ICAO_AEROPORTO_ORIGEM  345 non-null    string
dtypes: int32(1), string(4)
memory usage: 14.8 KB


In [228]:
teste_gb = teste.groupby(['MES_PARTIDA_PREVISTA', 'NOME_MES_PARTIDA_PREVISTA', 'SITUACAO_VOO'], observed=False).agg(
    TOTAL_SITUACAO=('SITUACAO_VOO', 'count')
)
teste_gb.sort_values(by=['MES_PARTIDA_PREVISTA'], inplace=True)
teste_gb.reset_index(inplace=True)
teste_gb

MES_PARTIDA_PREVISTA NOME_MES_PARTIDA_PREVISTA SITUACAO_VOO  \
0                      1                   janeiro    REALIZADO   
1                      2                 fevereiro    REALIZADO   
2                      3                     março    REALIZADO   
3                      4                     abril    REALIZADO   
4                      5                      maio    REALIZADO   
5                      6                     junho    REALIZADO   
6                      7                     julho    REALIZADO   
7                      8                    agosto    REALIZADO   
8                      9                  setembro    REALIZADO   
9                     10                   outubro    REALIZADO   
10                    11                  novembro    REALIZADO   
11                    12                  dezembro    REALIZADO   

    TOTAL_SITUACAO  
0               41  
1               20  
2               23  
3               21  
4               20  
5               22  
6               21  
7               23  
8               23  
9               24  
10              53  
11              54

In [229]:
fig = px.bar(
    teste_gb, 
    x='NOME_MES_PARTIDA_PREVISTA', 
    y='TOTAL_SITUACAO', 
    text_auto=True,
)
fig.update_traces(
    textposition='outside'
)
fig.show()

- Total de Vôos por aeroporto

In [230]:

colunas = ['NOME_AEROPORTO_ORIGEM', 'MES_PARTIDA_PREVISTA', 'NOME_MES_PARTIDA_PREVISTA', 'SIGLA_ICAO_AEROPORTO_ORIGEM']
teste = pd.read_parquet('../data/processed/df_final.parquet', columns=colunas)

In [231]:
teste.query('SIGLA_ICAO_AEROPORTO_ORIGEM == "SBRP"')

NOME_AEROPORTO_ORIGEM  MES_PARTIDA_PREVISTA NOME_MES_PARTIDA_PREVISTA  \
1929             LEITE LOPES                     1                   janeiro   
1930             LEITE LOPES                     1                   janeiro   
1931             LEITE LOPES                     1                   janeiro   
1932             LEITE LOPES                     1                   janeiro   
1933             LEITE LOPES                     1                   janeiro   
...                      ...                   ...                       ...   
707275           LEITE LOPES                     1                   janeiro   
707479           LEITE LOPES                     1                   janeiro   
707480           LEITE LOPES                     1                   janeiro   
707481           LEITE LOPES                     1                   janeiro   
710551           LEITE LOPES                     1                   janeiro   

       SIGLA_ICAO_AEROPORTO_ORIGEM  
1929                          SBRP  
1930                          SBRP  
1931                          SBRP  
1932                          SBRP  
1933                          SBRP  
...                            ...  
707275                        SBRP  
707479                        SBRP  
707480                        SBRP  
707481                        SBRP  
710551                        SBRP  

[4955 rows x 4 columns]

In [233]:
teste.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 717357 entries, 0 to 717356
Data columns (total 4 columns):
 #   Column                       Non-Null Count   Dtype   
---  ------                       --------------   -----   
 0   NOME_AEROPORTO_ORIGEM        717357 non-null  string  
 1   MES_PARTIDA_PREVISTA         717357 non-null  int32   
 2   NOME_MES_PARTIDA_PREVISTA    717357 non-null  category
 3   SIGLA_ICAO_AEROPORTO_ORIGEM  717357 non-null  string  
dtypes: category(1), int32(1), string(2)
memory usage: 14.4 MB


In [234]:
teste['NOME_MES_PARTIDA_PREVISTA'] = teste['NOME_MES_PARTIDA_PREVISTA'].astype('string')

In [239]:
total_voos =teste.groupby(['MES_PARTIDA_PREVISTA', 'NOME_MES_PARTIDA_PREVISTA']).agg(
    TOTAL_VOOS=('NOME_AEROPORTO_ORIGEM', 'count')
)
total_voos.reset_index(inplace=True)
total_voos['TOTAL_VOOS'] = total_voos['TOTAL_VOOS'].astype('int32')


In [240]:
total_voos

MES_PARTIDA_PREVISTA NOME_MES_PARTIDA_PREVISTA  TOTAL_VOOS
0                      1                   janeiro       78523
1                      2                 fevereiro       46597
2                      3                     março       54542
3                      4                     abril       52739
4                      5                      maio       58631
5                      6                     junho       55320
6                      7                     julho       63666
7                      8                    agosto       60677
8                      9                  setembro       58683
9                     10                   outubro       61976
10                    11                  novembro       60523
11                    12                  dezembro       65480

In [241]:
total_voos.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 12 entries, 0 to 11
Data columns (total 3 columns):
 #   Column                     Non-Null Count  Dtype 
---  ------                     --------------  ----- 
 0   MES_PARTIDA_PREVISTA       12 non-null     int32 
 1   NOME_MES_PARTIDA_PREVISTA  12 non-null     string
 2   TOTAL_VOOS                 12 non-null     int32 
dtypes: int32(2), string(1)
memory usage: 320.0 bytes


In [242]:
fig = px.bar(
    total_voos, 
    x='NOME_MES_PARTIDA_PREVISTA', 
    y='TOTAL_VOOS', 
    text_auto='0',
)
fig.update_traces(
    textposition='outside'
)
fig.show()

- Total de Vôos por compania e aeroporto por mês

In [67]:
base.columns

Index(['Unnamed: 0.1', 'Unnamed: 0', 'NOME_AEROPORTO_ORIGEM',
       'CIDADE_AEROPORTO_ORIGEM', 'UF_AEROPORTO_ORIGEM',
       'SIGLA_ICAO_EMPRESA_AEREA', 'NUMERO_VOO', 'CODIGO_TIPO_LINHA',
       'SIGLA_ICAO_AEROPORTO_ORIGEM', 'SIGLA_ICAO_AEROPORTO_DESTINO',
       'PARTIDA_PREVISTA', 'PARTIDA_REAL', 'CHEGADA_PREVISTA', 'CHEGADA_REAL',
       'SITUACAO_VOO', 'SIGLA_OACI', 'NOME_AEROPORTO_DESTINO',
       'CIDADE_AEROPORTO_DESTINO', 'UF_AEROPORTO_DESTINO', 'NOME_EMPRESA',
       'MES_PARTIDA_PREVISTA', 'NOME_MES_PARTIDA_PREVISTA', 'MES_PARTIDA_REAL',
       'NOME_MES_PARTIDA_REAL', 'MES_CHEGADA_PREVISTA',
       'NOME_MES_CHEGADA_PREVISTA', 'MES_CHEGADA_REAL',
       'NOME_MES_CHEGADA_REAL', 'MINUTOS_ATRASADO_PARTIDA',
       'MINUTOS_ATRASADO_CHEGADA', 'DIA_DA_SEMANA_PARTIDA_PREVISTA',
       'INDICE_SEMANA_PARTIDA_PREVISTA', 'DIA_DA_SEMANA_PARTIDA_REAL',
       'INDICE_SEMANA_PARTIDA_REAL', 'DIA_DA_SEMANA_CHEGADA_REAL',
       'DIA_DA_SEMANA_CHEGADA_PREVISTA', 'INDICE_SEMANA_CHEGADA_R

In [247]:
colunas = [
        
        'SIGLA_ICAO_AEROPORTO_ORIGEM',
        'MES_PARTIDA_PREVISTA',
        'NOME_MES_PARTIDA_PREVISTA',
        'NUMERO_VOO' ,
        'SIGLA_ICAO_EMPRESA_AEREA',
        'NOME_EMPRESA' 
]


teste = pd.read_parquet('../data/processed/df_final.parquet', columns=colunas)
teste = teste.query(
    ' SIGLA_ICAO_EMPRESA_AEREA == "TAM"  ' \
    ' and SIGLA_ICAO_AEROPORTO_ORIGEM == "SBRP"'\
)

In [248]:
teste.info()

<class 'pandas.core.frame.DataFrame'>
Index: 348 entries, 488380 to 680870
Data columns (total 6 columns):
 #   Column                       Non-Null Count  Dtype   
---  ------                       --------------  -----   
 0   SIGLA_ICAO_AEROPORTO_ORIGEM  348 non-null    string  
 1   MES_PARTIDA_PREVISTA         348 non-null    int32   
 2   NOME_MES_PARTIDA_PREVISTA    348 non-null    category
 3   NUMERO_VOO                   348 non-null    string  
 4   SIGLA_ICAO_EMPRESA_AEREA     348 non-null    string  
 5   NOME_EMPRESA                 348 non-null    string  
dtypes: category(1), int32(1), string(4)
memory usage: 15.7 KB


In [249]:
teste['NOME_MES_PARTIDA_PREVISTA'] = teste['NOME_MES_PARTIDA_PREVISTA'].astype('string')

In [250]:
teste.info()

<class 'pandas.core.frame.DataFrame'>
Index: 348 entries, 488380 to 680870
Data columns (total 6 columns):
 #   Column                       Non-Null Count  Dtype 
---  ------                       --------------  ----- 
 0   SIGLA_ICAO_AEROPORTO_ORIGEM  348 non-null    string
 1   MES_PARTIDA_PREVISTA         348 non-null    int32 
 2   NOME_MES_PARTIDA_PREVISTA    348 non-null    string
 3   NUMERO_VOO                   348 non-null    string
 4   SIGLA_ICAO_EMPRESA_AEREA     348 non-null    string
 5   NOME_EMPRESA                 348 non-null    string
dtypes: int32(1), string(5)
memory usage: 17.7 KB


In [251]:
teste.head()

SIGLA_ICAO_AEROPORTO_ORIGEM  MES_PARTIDA_PREVISTA  \
488380                        SBRP                     1   
488381                        SBRP                     1   
537779                        SBRP                     1   
537780                        SBRP                     1   
537781                        SBRP                     1   

       NOME_MES_PARTIDA_PREVISTA NUMERO_VOO SIGLA_ICAO_EMPRESA_AEREA  \
488380                   janeiro       3867                      TAM   
488381                   janeiro       3867                      TAM   
537779                   janeiro       3992                      TAM   
537780                   janeiro       3992                      TAM   
537781                   janeiro       3992                      TAM   

       NOME_EMPRESA  
488380          TAM  
488381          TAM  
537779          TAM  
537780          TAM  
537781          TAM

In [252]:
teste_gb = teste.groupby([ 'MES_PARTIDA_PREVISTA', 'NOME_MES_PARTIDA_PREVISTA']).agg(
    TOTAL_VOOS=('SIGLA_ICAO_EMPRESA_AEREA', 'count')
).reset_index()

In [253]:
teste_gb

MES_PARTIDA_PREVISTA NOME_MES_PARTIDA_PREVISTA  TOTAL_VOOS
0                      1                   janeiro          41
1                      2                 fevereiro          20
2                      3                     março          23
3                      4                     abril          21
4                      5                      maio          20
5                      6                     junho          22
6                      7                     julho          21
7                      8                    agosto          23
8                      9                  setembro          23
9                     10                   outubro          24
10                    11                  novembro          55
11                    12                  dezembro          55

In [254]:
fig = px.bar(
    teste_gb, 
    x='NOME_MES_PARTIDA_PREVISTA', 
    y='TOTAL_VOOS', 
    text_auto='0',
)
fig.update_traces(
    textposition='outside'
)
fig.show()

- Analise da variação percentual da SITUAÇÂO voo Filtrando por Aerporto origem por mês (deixe filtro facultativo por empresa)

In [255]:
base.columns

Index(['NOME_AEROPORTO_ORIGEM', 'CIDADE_AEROPORTO_ORIGEM',
       'UF_AEROPORTO_ORIGEM', 'SIGLA_ICAO_EMPRESA_AEREA', 'NUMERO_VOO',
       'CODIGO_TIPO_LINHA', 'SIGLA_ICAO_AEROPORTO_ORIGEM',
       'SIGLA_ICAO_AEROPORTO_DESTINO', 'PARTIDA_PREVISTA', 'PARTIDA_REAL',
       'CHEGADA_PREVISTA', 'CHEGADA_REAL', 'SITUACAO_VOO', 'SIGLA_OACI',
       'NOME_AEROPORTO_DESTINO', 'CIDADE_AEROPORTO_DESTINO',
       'UF_AEROPORTO_DESTINO', 'NOME_EMPRESA', 'MES_PARTIDA_PREVISTA',
       'NOME_MES_PARTIDA_PREVISTA', 'MES_PARTIDA_REAL',
       'NOME_MES_PARTIDA_REAL', 'MES_CHEGADA_PREVISTA',
       'NOME_MES_CHEGADA_PREVISTA', 'MES_CHEGADA_REAL',
       'NOME_MES_CHEGADA_REAL', 'MINUTOS_ATRASADO_PARTIDA',
       'MINUTOS_ATRASADO_CHEGADA', 'DIA_DA_SEMANA_PARTIDA_PREVISTA',
       'INDICE_SEMANA_PARTIDA_PREVISTA', 'DIA_DA_SEMANA_PARTIDA_REAL',
       'INDICE_SEMANA_PARTIDA_REAL', 'DIA_DA_SEMANA_CHEGADA_REAL',
       'DIA_DA_SEMANA_CHEGADA_PREVISTA', 'INDICE_SEMANA_CHEGADA_REAL',
       'INDICE_SEMANA_CH

In [256]:
colunas = [
    'SITUACAO_VOO',
    'MES_PARTIDA_PREVISTA',
    'NOME_MES_PARTIDA_PREVISTA',
    'SIGLA_ICAO_AEROPORTO_ORIGEM',
    'NOME_AEROPORTO_ORIGEM',
]

In [280]:
teste = pd.read_parquet('../data/processed/df_final.parquet', columns=colunas)

In [281]:
teste.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 717357 entries, 0 to 717356
Data columns (total 5 columns):
 #   Column                       Non-Null Count   Dtype   
---  ------                       --------------   -----   
 0   SITUACAO_VOO                 717357 non-null  category
 1   MES_PARTIDA_PREVISTA         717357 non-null  int32   
 2   NOME_MES_PARTIDA_PREVISTA    717357 non-null  category
 3   SIGLA_ICAO_AEROPORTO_ORIGEM  717357 non-null  string  
 4   NOME_AEROPORTO_ORIGEM        717357 non-null  string  
dtypes: category(2), int32(1), string(2)
memory usage: 15.1 MB


In [282]:
teste.head()

SITUACAO_VOO  MES_PARTIDA_PREVISTA NOME_MES_PARTIDA_PREVISTA  \
0    REALIZADO                     1                   janeiro   
1    REALIZADO                     1                   janeiro   
2    REALIZADO                     1                   janeiro   
3    REALIZADO                     1                   janeiro   
4    REALIZADO                     2                 fevereiro   

  SIGLA_ICAO_AEROPORTO_ORIGEM NOME_AEROPORTO_ORIGEM  
0                        SBAE         BAURU/AREALVA  
1                        SBAE         BAURU/AREALVA  
2                        SBAE         BAURU/AREALVA  
3                        SBAE         BAURU/AREALVA  
4                        SBAE         BAURU/AREALVA

In [283]:
teste.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 717357 entries, 0 to 717356
Data columns (total 5 columns):
 #   Column                       Non-Null Count   Dtype   
---  ------                       --------------   -----   
 0   SITUACAO_VOO                 717357 non-null  category
 1   MES_PARTIDA_PREVISTA         717357 non-null  int32   
 2   NOME_MES_PARTIDA_PREVISTA    717357 non-null  category
 3   SIGLA_ICAO_AEROPORTO_ORIGEM  717357 non-null  string  
 4   NOME_AEROPORTO_ORIGEM        717357 non-null  string  
dtypes: category(2), int32(1), string(2)
memory usage: 15.1 MB


In [284]:
teste[['SITUACAO_VOO', 'NOME_MES_PARTIDA_PREVISTA']] = teste[['SITUACAO_VOO', 'NOME_MES_PARTIDA_PREVISTA']].astype('string')

In [285]:
teste = teste.query('SIGLA_ICAO_AEROPORTO_ORIGEM == "SBRP"  and MES_PARTIDA_PREVISTA == 1 ')
teste

SITUACAO_VOO  MES_PARTIDA_PREVISTA NOME_MES_PARTIDA_PREVISTA  \
1929      REALIZADO                     1                   janeiro   
1930      REALIZADO                     1                   janeiro   
1931      REALIZADO                     1                   janeiro   
1932      REALIZADO                     1                   janeiro   
1933      REALIZADO                     1                   janeiro   
...             ...                   ...                       ...   
707275    REALIZADO                     1                   janeiro   
707479    REALIZADO                     1                   janeiro   
707480    REALIZADO                     1                   janeiro   
707481    REALIZADO                     1                   janeiro   
710551    REALIZADO                     1                   janeiro   

       SIGLA_ICAO_AEROPORTO_ORIGEM NOME_AEROPORTO_ORIGEM  
1929                          SBRP           LEITE LOPES  
1930                          SBRP           LEITE LOPES  
1931                          SBRP           LEITE LOPES  
1932                          SBRP           LEITE LOPES  
1933                          SBRP           LEITE LOPES  
...                            ...                   ...  
707275                        SBRP           LEITE LOPES  
707479                        SBRP           LEITE LOPES  
707480                        SBRP           LEITE LOPES  
707481                        SBRP           LEITE LOPES  
710551                        SBRP           LEITE LOPES  

[605 rows x 5 columns]

In [286]:
teste_gb =teste.groupby(['NOME_MES_PARTIDA_PREVISTA', 'SITUACAO_VOO', 'MES_PARTIDA_PREVISTA']).agg(
    TOTAL_VOOS=('SITUACAO_VOO', 'count')
)

teste_gb.reset_index(inplace=True)
teste_gb.sort_values(by=['MES_PARTIDA_PREVISTA'], inplace=True)
teste_gb['TOTAL_VOOS'] = teste_gb['TOTAL_VOOS'].astype('int32')

In [287]:
teste_gb.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2 entries, 0 to 1
Data columns (total 4 columns):
 #   Column                     Non-Null Count  Dtype 
---  ------                     --------------  ----- 
 0   NOME_MES_PARTIDA_PREVISTA  2 non-null      string
 1   SITUACAO_VOO               2 non-null      string
 2   MES_PARTIDA_PREVISTA       2 non-null      int32 
 3   TOTAL_VOOS                 2 non-null      int32 
dtypes: int32(2), string(2)
memory usage: 176.0 bytes


In [288]:
teste_gb

NOME_MES_PARTIDA_PREVISTA SITUACAO_VOO  MES_PARTIDA_PREVISTA  TOTAL_VOOS
0                   janeiro    CANCELADO                     1          21
1                   janeiro    REALIZADO                     1         584

- Analise da variação percentual da SITUAÇÂO voo Filtrando por Aerporto origem por mês compare com o mês anterior (Talvez com vísual cartões) deixe filtro facultativo por empresa)

- Compare total de vôos realizado no fim de semana em relação por dia útil por mês

<p> - Analise a FAIXA_ATRASO_PARTIDA por aeroporto e empresa por dia de semana e por mês </p>
<p> O volume de atraso no fim de semana, é maior ou menor em relação ao dia atual</p>
<p> Pode ser análisado a incidencia por aeroporto também</p>
<p>Detalhe por semana também</p>
<p>Eu tenho 3000 vôos que partiram no horário 50%(1500) da empresa A e 50% da empresa B</p>


<p> - Analise a FAIXA_ATRASO_CHEGADA por aeroporto e empresa por dia de semana e por mês </p>
<p> O volume de atraso no fim de semana, é maior ou menor em relação ao dia atual</p>
<p> Pode ser análisado a incidencia por aeroporto também</p>
<p>Detalhe por semana também</p>
<p>Eu tenho 3000 vôos que partiram no horário 50%(1500) da empresa A e 50% da empresa B</p>


<p> Ranking por FAIXA_ATRASO_CHEGADA selecionando aeroporto  por mês e empresa</p>

<p> Ranking por FAIXA_ATRASO_SAIDA selecionando aeroporto em empresa por mês</p>

<p> Quantifique a porcentagem por FAIXA_ATRASO_CHEGADA selecionando aeroportoa por mês</p>

<p> Quantifique a porcentagem por FAIXA_ATRASO_PARTIFA selecionando aeroporto  por mês</p>